### PR TITLE
EWPP-270: Add "Details" field group to news content type.

### PR DESCRIPTION
**Pull request by yenyasinn**
### Description

Added Details field group with new field Reference to the News default view mode.

### CODE REVIEW

File: modules/oe_theme_content_news/oe_theme_content_news.info.yml [hunk 1]
```diff
@@ -7,4 +7,4 @@ dependencies:
   - extra_field:extra_field
   - oe_content_news:oe_content_news
   - oe_theme:oe_theme_helper
-
+  - ui_patterns:ui_patterns_field_group
```
Comment (ademarco) [inline modules/oe_theme_content_news/oe_theme_content_news.info.yml:10]: We need to add a post update that enables this new dependency.

**Review by ademarco — CHANGES_REQUESTED** [review]

File: modules/oe_theme_content_news/oe_theme_content_news.info.yml (hunk 1 above)
Comment (yenyasinn) [reply modules/oe_theme_content_news/oe_theme_content_news.info.yml:10]: Implemented in oe_theme_content_news_post_update_00004()

File: templates/field/field--node--oe-author--oe-news.html.twig [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{#
+/**
+ * @file
+ * News author field template.
+ *
+ * @see ./core/themes/stable/templates/field/field.html.twig
+ */
+#}
+{% for item in items %}{{ item.content }}{{ not loop.last ? ' | ' }}{% endfor %}
```
Comment (yenyasinn) [inline templates/field/field--node--oe-author--oe-news.html.twig:9]: I decided to write it on one line to get rid of new line characters. It is helpful to write test cases for FieldListAssert.

**Review by ademarco — DISMISSED** [review]

File: tests/Functional/ContentNewsRenderTest.php [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    // oe_theme itself.
+    \Drupal::service('plugin.manager.ui_patterns')->clearCachedDefinitions();
+
+    Role::load(RoleInterface::ANONYMOUS_ID)
+      ->grantPermission('view published skos concept entities')
+      ->save();
+  }
+
+  /**
+   * Tests that the Project page renders correctly.
```
Comment (drishu) [inline tests/Functional/ContentNewsRenderTest.php]: Tests that the News page renders correctly.

File: tests/Functional/ContentNewsRenderTest.php [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+   */
+  public function testNewsRendering(): void {
+    // Create a Project node.
```
Comment (drishu) [inline tests/Functional/ContentNewsRenderTest.php]: Create a News node.

File: tests/Functional/ContentNewsRenderTest.php [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    /** @var \Drupal\node\Entity\Node $node */
+    $node = $this->getStorage('node')->create([
+      'type' => 'oe_news',
+      'title' => 'Test news node',
+      'oe_teaser' => 'News teaser',
```
Comment (drishu) [inline tests/Functional/ContentNewsRenderTest.php:57]: lets also test rendering of oe_news_featured_media, oe_related_links

**Review by drishu — DISMISSED** [review]

File: tests/Functional/ContentNewsRenderTest.php (hunk 5 above)
Comment (yenyasinn) [reply tests/Functional/ContentNewsRenderTest.php:57]: The aim of the ticket is to add Reference field and theme Details group.

Comment (drishu) [reply tests/Functional/ContentNewsRenderTest.php:57]: if it will be in a follow up then its ok

Comment (ademarco) [reply tests/Functional/ContentNewsRenderTest.php:57]: Yeah we will merge the two tests later on, depending which issue gets merged first, this one or #583

File: modules/oe_theme_content_news/oe_theme_content_news.module [hunk 6]
```diff
@@ -29,3 +31,16 @@ function oe_theme_content_news_preprocess_node__oe_news(&$variables) {
...
+ * Implements hook_ENTITY_TYPE_view_alter().
+ */
+function oe_theme_content_news_node_view_alter(array &$build, EntityInterface $entity, EntityViewDisplayInterface $display) {
+  if ($entity->bundle() === 'oe_news'
+    && $build['#view_mode'] === 'full'
+    && !$entity->get('oe_author')->isEmpty()
+    && $entity->get('oe_author')->count() > 1
+    && isset($build['oe_author']['#title'])) {
+    $build['oe_author']['#title'] = t('Authors');
+  }
```
Comment (ademarco) [inline modules/oe_theme_content_news/oe_theme_content_news.module:54]: Let's use the same way we do on other alters, even if it's one field for now (more will come here):

```php
  if ($entity->bundle() !== 'oe_news') {
    return;
  }

  // Field labels are different in frontend and backend.
  // Here we override them, per view mode.
  if ($build['#view_mode'] === 'full') {
    $plural_label_overrides = [
      'oe_author' => t('Authors'),
    ];

    foreach ($plural_label_overrides as $name => $label_override) {
      if (!$entity->get($name)->isEmpty() && $entity->get($name)->count() > 1 && isset($build[$name]['#title'])) {
        $build[$name]['#title'] = $label_override;
      }
    }
  }
```

**Review by ademarco — CHANGES_REQUESTED** [review]

**Review by imanoleguskiza — DISMISSED** [review]

**Review by imanoleguskiza — APPROVED** [review]

**Review by ademarco — APPROVED** [review]

**Pull request merged**